### DISCUSSION TITLE
[Feature] <Plugin Support>

### DISCUSSION THREAD

**Discussion by Tmatz27** [immich-app/immich#8173] [General]
### The feature

It would be nice if we could add Plugins into Immich. I would like to add additional AI object detection or facial recognition plugins. Ive found the baked in facial recognition is not up to par with other services already provided. I have a folder full of people and not a single person is detected currently. 

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by alextran1502** [comment]
This seems like an issue with the machine learning on your machine rather than the model Immich uses, which has been very good. Maybe provide us with more info so that we can help

**Comment by Tmatz27** [comment]
Not sure where to look to provide more data. there is no config.yml or log file in the docker appdata folder, and if i click on log on the docker image it only prints out a single error. Ive ran face detection a few times and it always returned zero results in a matter of seconds even though its searching over 15000 images. 

Ill attached my docker setup, but please advise on what else i should include.
![Screenshot 2024-03-21 160029](https://github.com/immich-app/immich/assets/116435680/a8555180-e3c3-4ce8-95e6-d0fa065cbd7f)
![Screenshot 2024-03-21 160046](https://github.com/immich-app/immich/assets/116435680/a96e245d-a29c-4ad4-95fb-70838b2922f5)

**Reply by alextran1502** [reply]
A log to the container would be helpful!

**Reply by Tmatz27** [reply]
I haven't been able to find a single log file or config file in the entire unraid immich app data folder. No idea where it would be. 

When I go into the immich file there is just a machine learning folder and I've gone through the entire folder structure and couldn't find anything. 
Is there a variable that I can add for debugging or to create the log file?

**Reply by alextran1502** [reply]
I don't use unraid template so I am not sure, you might have to ask the community for help

**Comment by Tmatz27** [comment]
Sounds good thank you for the help

**Comment by DrissiReda** [comment]
I agree that the idea is good, even if not necessarily for AI specific models. What I had in mind was something like Immich Power tools instead.
